Mathlib/RingTheory/PowerSeries/Substitution.lean

```diff
@@ -0,0 +1,381 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.LinearAlgebra.Finsupp.Pi
+import Mathlib.RingTheory.MvPowerSeries.Substitution
+import Mathlib.RingTheory.PowerSeries.Evaluation
+import Mathlib.RingTheory.PowerSeries.Basic
+
+import Mathlib.RingTheory.PowerSeries.Order
+
+/-! # Substitutions in power series
+
+A (possibly multivariate) power series can be substituted into
+a (univariate) power series if and only if its constant coefficient is nilpotent.
+
+This is a particularization of the substitution of multivariate power series
+to the case of univariate power series.
+
+-/
+
+namespace PowerSeries
+
+variable
+  {A : Type*} [CommRing A]
+  {R : Type*} [CommRing R] [Algebra A R]
+  {τ : Type*}
+  {S : Type*} [CommRing S]
+
+open MvPowerSeries.WithPiTopology
+
+attribute [local instance] DiscreteTopology.instContinuousSMul
+
+/-- Families of power series which can be substituted in other power series -/
+abbrev HasSubst (a : MvPowerSeries τ S) : Prop :=
```

Comment:
This (and the following results) seems to involve `MvPowerSeries` and it doesn't mention `PowerSeries`. Should this go somewhere else? If I understand correctly, the difference wrt to [MvPowerSeries.HasSubst](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/MvPowerSeries/Substitution.html#MvPowerSeries.HasSubst) is that here there is only one power series (in particular the doc should be changed).

Reply:
Yes : the point is that one substitutes a single power series, but that it may have several indeterminates.
This is related to the discussion with @b-mehta above.
I didn't believe at first that there should be four API for substituting (Mv)PowerSeries into (Mv)PowerSeries, but I may be wrong.

Reply:
I didn't follow the whole discussion, so sorry if this has been already explained, but do we really want an API to substitute a multivariate power series into a power series?

We of course need the general theory for `MvPowerSeries`, that is already there, and I perfectly understand that we want to be able to play with `PowerSeries` without mentioning `Unit`, but substituting a multivariate power series into a power series seems a pretty rare thing to do.